pkg/kubelet/kubelet_pods.go

```diff
@@ -465,7 +469,9 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *v1.Pod, container *v1.Contai
 	}
 	opts.Envs = append(opts.Envs, envs...)
 
-	mounts, cleanupAction, err := makeMounts(pod, kl.getPodDir(pod.UID), container, hostname, hostDomainName, podIP, volumes, kl.hostutil, kl.subpather, opts.Envs)
+	// we can only mount individual files (e.g.: /etc/hosts, termination-log files) on Windows only if we're using Containerd.
+	isContainerd := kl.containerRuntime.Type() == kubetypes.ContainerdContainerRuntime
```

Comment:
where is this set? The kubelet config / command line still just includes a remote runtime.

I'm wondering if it makes more sense to implement a function that is capability focused, rather than returning the runtime

```
func SupportsSingleFileMapping() bool {
    switch (runtime.GOOS) {
       "windows": {
           return kl.containerRuntime.Type() == kubetypes.RemoteContainerRuntime
       }
       default: {
           return true
       }
    }
}
```